MediaMTX crashes after enabling recording on path with source on demand

## Which version are you using?

v1.7.0

## Which operating system are you using?

- [x] Linux amd64 Docker

## Describe the issue

MediaMTX crashes after sending PATCH request that enables recording for path with `sourceOnDemand` true.
Expected behavior should be automatically changing `sourceOnDemand` from value `true` to `false`, when we send `{ "record": true }` request body. Another option is returning exception from API with message saying something like "you cannot start recording on camera with source on demand".

## Describe how to replicate the issue

1. Start the server.
2. Add path with properties `source`, `sourceOnDemand` (true) and `sourceOnDemandCloseAfter` (through API).
3. Send PATCH request with body `{ "record": true }` through API.
4. Check logs to see the segmentation error.

## Did you attach the server logs?

yes

Below, after sending PATCH request.
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xcf1d60]

goroutine 193 [running]:
github.com/bluenviron/mediamtx/internal/core.(*staticSourceHandler).reloadConf(0x75ce25?, 0xc00051e540?)
        /s/internal/core/static_source_handler.go:229 +0x20
created by github.com/bluenviron/mediamtx/internal/core.(*path).doReloadConf in goroutine 164
        /s/internal/core/path.go:374 +0x173
```

## Did you attach a network dump?

no

--- a/internal/core/path.go
+++ b/internal/core/path.go
@@ -371,7 +371,7 @@ func (pa *path) doReloadConf(newConf *conf.Path) {
 	pa.confMutex.Unlock()
 
 	if pa.conf.HasStaticSource() {
-		go pa.source.(*staticSourceHandler).reloadConf(newConf)
+		pa.source.(*staticSourceHandler).reloadConf(newConf)
 	}
 
 	if pa.conf.Record {

--- a/internal/core/static_source_handler.go
+++ b/internal/core/static_source_handler.go
@@ -224,10 +224,18 @@ func (s *staticSourceHandler) run() {
 }
 
 func (s *staticSourceHandler) reloadConf(newConf *conf.Path) {
-	select {
-	case s.chReloadConf <- newConf:
-	case <-s.ctx.Done():
+	ctx := s.ctx
+
+	if !s.running {
+		return
 	}
+
+	go func() {
+		select {
+		case s.chReloadConf <- newConf:
+		case <-ctx.Done():
+		}
+	}()
 }
 
 // APISourceDescribe instanceements source.

